File: backend/src/lib/product/Product.ts
import DeliveryRule from "./DeliveryRule";
import PricingRule from "./PricingRule";
import QuantityPricing from "./QuantityPricing";

// Models
export default class Product {
  name: string;
  pricingRules: PricingRule[];
  deliveryRules: DeliveryRule[];
  quantityPricings: QuantityPricing[];

  constructor(
    name: string,
    pricingRules: PricingRule[] = [],
    deliveryRules: DeliveryRule[] = [],
    quantityPricings: QuantityPricing[] = []
  ) {
    this.name = name;
    this.pricingRules = pricingRules;
    this.deliveryRules = deliveryRules;
    this.quantityPricings = quantityPricings;
  }
}

File: price-engine/app/lib/product/product.py
from dataclasses import dataclass, field
from typing import List
from .pricing_rule import PricingRule
from .delivery_rule import DeliveryRule
from .quantity_pricing import QuantityPricing

@dataclass
class Product:
    name: str
    pricing_rules: List[PricingRule] = field(default_factory=list)
    delivery_rules: List[DeliveryRule] = field(default_factory=list)
    quantity_pricings: List[QuantityPricing] = field(default_factory=list)
